sympy/ntheory/primetest.py

```diff
@@ -92,8 +92,9 @@ def is_square(n, prep=True):
         m = n % 63
         if not ((m*0x3d491df7) & (m*0xc824a9f9) & 0x10f14008):
             from sympy.ntheory import perfect_power
-            if perfect_power(n, [2]):
-                return True
+            pp = perfect_power(n, [2])
+            if pp:
+                return pp[1] % 2 == 0
```

Comment:
Is it better to do this or pass `big=False` to `perfect_power()`?

Reply:
Doesn't work- you still get `(60, 3)`.

Reply:
I meant big=False and p[1] == 2

Reply:
I don't actually know which is better. Presumably one is more performant.

Reply:
Doing a naive test with the integers in `range(1, 10**7)` I get 18.89302659034729 seconds for your version with `big=False` and 19.271745443344116 seconds with the mod 2 version- not a huge difference.

Reply:
In some cases, `big=False` is slower
```python
>>> cProfile.run('perfect_power((2*3*5*7)**1000000, big=True)[1] % 2 == 0')
         54 function calls in 2.616 seconds
>>> cProfile.run('perfect_power((2*3*5*7)**1000000, big=False)[1] == 2')
         81 function calls in 2.907 seconds
```